### PR TITLE
sklearnserver/xgbserver version upgrade

### LINKED ISSUE #1791: Failed model server becomes ready
/kind bug

**What steps did you take and what happened:**

When XGBserver fails to load the model, the inference service still gives the inference service the status "Ready":

```
/usr/local/lib/python3.7/site-packages/ray/autoscaler/_private/cli_logger.py:61: FutureWarning: Not all Ray CLI dependencies were found. In Ray 1.4+, the Ray CLI, autoscaler, and dashboard will only be usable via `pip install 'ray[default]'`. Please update your install command.
"update your install command.", FutureWarning)
[I 210906 08:11:56 storage:50] Copying contents of /mnt/models to local
[E 210906 08:11:56 __main__:45] fail to load model xe57583ad-cae3-44ae-aa73-7da0e129cfe4 from dir /mnt/models. exception type <class 'xgboost.core.XGBoostError'>, exception msg: b'[08:11:56] /workspace/dmlc-core/src/io/local_filesys.cc:209: Check failed: allow_null LocalFileSystem::Open "/mnt/models/model.bst": No such file or directory\n\nStack trace returned 10 entries:\n[bt] (0) /usr/local/xgboost/libxgboost.so(dmlc::StackTrace(unsigned long)+0x47) [0x7f5cc50c3fc7]\n[bt] (1) /usr/local/xgboost/libxgboost.so(dmlc::io::LocalFileSystem::Open(dmlc::io::URI const&, char const*, bool)+0x411) [0x7f5cc5267dd1]\n[bt] (2) /usr/local/xgboost/libxgboost.so(dmlc::Stream::Create(char const*, char const*, bool)+0x3a) [0x7f5cc52146fa]\n[bt] (3) /usr/local/xgboost/libxgboost.so(XGBoosterLoadModel+0x27) [0x7f5cc50ba717]\n[bt] (4) /usr/lib/x86_64-linux-gnu/libffi.so.7(+0x6d1d) [0x7f5cedfcad1d]\n[bt] (5) /usr/lib/x86_64-linux-gnu/libffi.so.7(+0x6289) [0x7f5cedfca289]\n[bt] (6) /usr/local/lib/python3.7/lib-dynload/_ctypes.cpython-37m-x86_64-linux-gnu.so(+0xd679) [0x7f5cedfdd679]\n[bt] (7) /usr/local/lib/python3.7/lib-dynload/_ctypes.cpython-37m-x86_64-linux-gnu.so(+0x12c2a) [0x7f5cedfe2c2a]\n[bt] (8) /usr/local/lib/libpython3.7m.so.1.0(_PyObject_FastCallKeywords+0x95) [0x7f5cee9d9eb5]\n[bt] (9) /usr/local/lib/libpython3.7m.so.1.0(+0x165451) [0x7f5ceea36451]\n\n'
[I 210906 08:11:56 kfserver:121] Setting asyncio max_workers as 8
[I 210906 08:11:56 kfserver:128] Listening on port 8080
[I 210906 08:11:56 kfserver:130] Will fork 1 workers
[W 210906 08:12:30 web:2243] 404 POST /v1/models/xe57583ad-cae3-44ae-aa73-7da0e129cfe4:predict (127.0.0.1) 1.18ms
```

**What did you expect to happen:**

The service should raise an error and exit with exit code 1. This would let the controller manager know the deployment of the model failed, and set the status Ready to `false`.

**Environment:**

- Istio Version: 1.4.10
- Knative Version: 0.18.0
- KFServing Version: 0.5.1
- Kubernetes version: (use `kubectl version`): 1.18

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
upgrade sklearn to 1.0.1 and xgboost version to 1.5.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1791 

**Special notes for your reviewer**:
other improvements include:
1. look for models from model repository only if the model file is missing from the parent directory
2. updated test model artifacts with sklearn 1.0.1 and xgboost 1.5.0 version
3. Support predict_proba for sklearnserver

**Release note**:

```release-note
- Upgrade scikit-learn to 1.0.1 for sklearnserver
- Update xgboost to 1.5.0 for xgbserver
- Update alibi to 0.6.2
```

### CODE REVIEW

**Comment by yuzisun** [comment]
@andyi2it @markwinter @Suresh-Nakkeran Can you help take a look ?

File: python/xgbserver/xgbserver/xgboost_model_repository.py [hunk 1]
```diff
@@ -20,6 +20,12 @@ class XGBoostModelRepository(KFModelRepository):
     def __init__(self, model_dir: str = MODEL_MOUNT_DIRS, nthread: int = 1):
         super().__init__(model_dir)
         self.nthread = nthread
+        for name in os.listdir(model_dir):
```
Comment (markwinter) [inline python/xgbserver/xgbserver/xgboost_model_repository.py]: This bit of code is quite useful and used in a few locations.

Should it be a method on `KFModelRepository`? Something like `load_models_from_dir(dir: string)`

**Comment by markwinter** [comment]
@yuzisun 
Looks good to me.
Thanks for organising all the package requirements and versions

**Comment by andyi2it** [comment]
@yuzisun 
Looks good to me. Have my doubts on model ready issue #1791 
Will test it locally and update here.

**Comment by tuan-ff** [comment]
@yuzisun what is the image tag or `defaultImageVersion` we should use for this upgrade in `inferenceservice-config`? Thanks!

**Pull request merged**